src/main/java/ch/njol/skript/sections/EffSecSpawn.java

```diff
@@ -142,16 +146,19 @@ protected TriggerItem walk(Event e) {
 				SpawnEvent spawnEvent = new SpawnEvent(o);
 				// Copy the local variables from the calling code to this section
 				Variables.setLocalVariables(spawnEvent, localVars);
-				trigger.execute(spawnEvent);
+				TriggerItem.walk(trigger, spawnEvent);
+				Variables.setLocalVariables(event, Variables.copyLocalVariables(spawnEvent));
+				// Clear spawnEvent's local variables as it won't be done automatically
+				Variables.removeLocals(spawnEvent);
 			};
 		} else {
 			consumer = null;
 		}
 
-		Number a = amount != null ? amount.getSingle(e) : 1;
+		Number a = amount != null ? amount.getSingle(event) : 1;
 		if (a != null) {
-			EntityType[] ts = types.getArray(e);
-			for (Location l : locations.getArray(e)) {
+			EntityType[] ts = types.getArray(event);
+			for (Location l : locations.getArray(event)) {
```

Comment:
We can use amount = this.amount, types = this.types, location : this.locations appropriately.